### PR TITLE
GUVNOR-3105: [Guided Decision Table] Improve experience of using keyboard in Decision Table

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/GUVNOR-3105

### CODE REVIEW

**Comment by jomarko** [comment]
jenkins execute full downstream build

**Comment by manstis** [comment]
jenkins execute full downstream build

**Comment by manstis** [comment]
jenkins execute full downstream build

**Comment by manstis** [comment]
@jomarko Fixed the ```Shift-tab``` issue.

**Comment by jomarko** [comment]
@manstis thanks, and what about others, are you not able to reproduce them?

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/table/keyboard/BaseKeyboardTest.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    @Mock
+    private GridColumnRenderer columnRenderer;
+
+    @Mock
+    private BaseGridRendererHelper rendererHelper;
+
+    protected GridData uiModel = new BaseGridData();
+
+    @Before
+    @SuppressWarnings("unchecked")
```
Comment (karreiro) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/table/keyboard/BaseKeyboardTest.java]: I think that this warning can be removed by declaring the `columnRenderer` like this:
```java
private GridColumnRenderer<?> columnRenderer
```

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/table/keyboard/BaseKeyboardOperationTest.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import org.uberfire.ext.wires.core.grids.client.widget.layer.GridLayer;
+
+import static org.junit.Assert.*;
+
+public abstract class BaseKeyboardOperationTest<H extends KeyboardOperation> extends BaseKeyboardTest {
+
+    protected H handler;
+
+    @Before
+    @SuppressWarnings("unchecked")
```
Comment (karreiro) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/table/keyboard/BaseKeyboardOperationTest.java]: I think that you can remove this line :-)

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/keyboard/GotoBeginningOfRow.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        return TriStateBoolean.FALSE;
+    }
+
+    @Override
+    public TriStateBoolean isControlKeyDown() {
+        return TriStateBoolean.FALSE;
+    }
+
+    @Override
+    @SuppressWarnings("unused")
```
Comment (karreiro) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/keyboard/GotoBeginningOfRow.java]: I think that you can remove this line :-)

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/keyboard/GotoEndOfRow.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        return TriStateBoolean.FALSE;
+    }
+
+    @Override
+    public TriStateBoolean isControlKeyDown() {
+        return TriStateBoolean.FALSE;
+    }
+
+    @Override
+    @SuppressWarnings("unused")
```
Comment (karreiro) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/keyboard/GotoEndOfRow.java]: I think that you can remove this line :-)

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/keyboard/SelectionCopy.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        return KeyCodes.KEY_C;
+    }
+
+    @Override
+    public TriStateBoolean isControlKeyDown() {
+        return TriStateBoolean.TRUE;
+    }
+
+    @Override
+    @SuppressWarnings("unused")
```
Comment (karreiro) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/keyboard/SelectionCopy.java]: I think that you can remove this line :-)

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/keyboard/SelectionCut.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        return KeyCodes.KEY_X;
+    }
+
+    @Override
+    public TriStateBoolean isControlKeyDown() {
+        return TriStateBoolean.TRUE;
+    }
+
+    @Override
+    @SuppressWarnings("unused")
```
Comment (karreiro) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/keyboard/SelectionCut.java]: I think that you can remove this line :-)

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/keyboard/SelectionPaste.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        return KeyCodes.KEY_V;
+    }
+
+    @Override
+    public TriStateBoolean isControlKeyDown() {
+        return TriStateBoolean.TRUE;
+    }
+
+    @Override
+    @SuppressWarnings("unused")
```
Comment (karreiro) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/keyboard/SelectionPaste.java]: I think that you can remove this line :-)

**Comment by manstis** [comment]
@karreiro Thanks. Updates made.

@jomarko I could not replicate any of the reported issues (other than ```Shift-tab```); but I do have the V&V panel disabled - that's why I asked about "focus" on the JIRA. Let me re-enable it and check.

**Comment by manstis** [comment]
@jomarko Fixed ```Shift-Home``` and ```Shift-End``` operation for when "first loaded".

**Comment by manstis** [comment]
@jomarko OK.. All fixed. Key press handlers are no longer registered on the canvas.

**Comment by jomarko** [comment]
jenkins execute full downstream build

**Comment by manstis** [comment]
@karreiro Hi, could you please check my changes and approve?

**Pull request merged**